xtext/org.icyphy.linguafranca/src/lib/core/federate.c

```diff
@@ -149,31 +295,269 @@ void send_via_rti_timed(unsigned int port, unsigned int federate, size_t length,
  *  @param time The time of this federate's next event.
  */
 void send_time(unsigned char type, instant_t time) {
+    DEBUG_PRINT("Sending time %lld to the RTI.", time);
     unsigned char buffer[9];
     buffer[0] = type;
     encode_ll(time, &(buffer[1]));
-    write_to_socket(rti_socket, 9, buffer);
+    write_to_socket(_lf_rti_socket, 9, buffer, "Federate %d failed to send time to the RTI.", _lf_my_fed_id);
 }
 
 /** Send a STOP message to the RTI, which will then broadcast
  *  the message to all federates.
  *  This function assumes the caller holds the mutex lock.
  */
 void __broadcast_stop() {
-    printf("Federate %d requesting a whole program stop.\n", __my_fed_id);
+    DEBUG_PRINT("Federate %d requesting a whole program stop.\n", _lf_my_fed_id);
     send_time(STOP, current_time);
 }
 
+/**
+ * Thread to accept connections from other federates that send this federate
+ * messages directly (not through the RTI). This thread starts a thread for
+ * each accepted socket connection and then waits for all those threads to exit
+ * before exiting itself.
+ * @param ignored No argument needed for this thread.
+ */
+void* handle_p2p_connections_from_federates(void *ignored) {
+    int received_federates = 0;
+    pthread_t thread_ids[_lf_number_of_inbound_physical_connections];
+    while (received_federates < _lf_number_of_inbound_physical_connections) {
+        // Wait for an incoming connection request.
+        struct sockaddr client_fd;
+        uint32_t client_length = sizeof(client_fd);
+        int socket_id = accept(_lf_server_socket, &client_fd, &client_length);
+        // FIXME: Error handling here is too harsh maybe?
+        if (socket_id < 0) {
+            return NULL;
```

Comment:
Print/log error message?

Reply:
This should be fixed in 755c2d77108667ba5bc22ff800fd94ef3fac9dd9. I don't know why GitHub is not showing this comment as outdated.